src/engine/Match_rules.ml

```diff
@@ -161,8 +161,14 @@ let per_rule_boilerplate_fn ~timeout ~timeout_threshold =
 (* Entry point *)
 (*****************************************************************************)
 
-let check ~match_hook ~timeout ~timeout_threshold (xconf : Match_env.xconfig)
-    rules xtarget =
+let check ~match_hook ~timeout ~timeout_threshold
+    ?(dependency_match_table : Match_dependency.dependency_match_table option)
```

Comment:
do we need to pass down the dependency stuff? Could we just post-process the matches instead and
check for the dependency in a separate file?

Reply:
added a comment